Automattic/wp-calypso
# CPT: Include all available post actions as independent menu item components

This pull request seeks to implement most all post actions to be available for custom post types and adds previously unaccounted-for capabilities checking and labels as derived from the post type object.

Specifically, actions include: "View", "Publish", "Edit", "Restore", "Trash" / "Permanently Delete"

![Actions](https://cloud.githubusercontent.com/assets/1779930/16469104/373213c2-3e1d-11e6-8d2a-3d5d0728e80f.png)

__Implementation notes:__

As I started fleshing out `<PostTypeListActions />` with individual action buttons, I found that the component became quite large with logic and props. It therefore made sense to me to instead allow the action menu items to be independent, with their own rendering behavior and click handling.

`<PopoverMenuItem />` has been enhanced to allow rendering as an anchor tag (link), and to accept props not otherwise listed in the `propTypes` (e.g. `target`).

__Testing instructions:__

Verify that post actions on the [custom post types list screen](http://calypso.localhost:3000/types/post) are consistent with the status and type of the post, and that actions behave as you'd expect (with the exception of Restore, which is to be implemented in a future pull request). Verify that actions are not made available if your user account does not have permission to perform that action.

Test live: https://calypso.live/?branch=add/cpt-actions
